teamfoundationcontrol.cpp

```diff
@@ -231,3 +249,36 @@ Core::ShellCommand *TeamFoundation::Internal::TeamFoundationControl::createIniti
     command->addJob(settings.binaryPath(), getArgs, -1);
     return command;
 }
+
+////////////////////////////////////////////////////////////////////////////////
+void TeamFoundationControl::DirectoryTree::Add(const QString& directory)
+{
+    for (int i = 0; i < m_entries.size(); ++i) {
+        // candidate shadowed?
+        if (IsParentOrSame(m_entries[i], directory)) {
+            return;
+        }
+
+        // candidate shadows?
+        if (IsParentOrSame(directory, m_entries[i])) {
+            m_entries[i] = m_entries[m_entries.size()-1];
+            m_entries.pop_back();
+            --i;
+        }
+    }
+
+    m_entries << directory;
+}
+
+bool TeamFoundationControl::DirectoryTree::IsParentOrSame(const QString& lsh, const QString& rhs)
+{
+    return rhs.startsWith(lsh, Qt::CaseInsensitive);

```

Comment:
I am still not sure I follow, take the following example:
the directory that is managed is everything in c:/test/
user tries to access file in folder c:/test_test/
This would result in the following execution in case of no slashes
lhs: C:/test
rhs: C:/test_test
As far as I can see, this would cause 
Utils::contains(m_entries, [p] (const QString& entry) { return IsParentOrSame(entry, *p); });
to return true, as C:/test_test does start with C:/test, even though only C:/test is managed